### DISCUSSION TITLE
Intel 82599 NIC fail to boot iPXE chainloading

### DISCUSSION THREAD

**Discussion by JetYu1234** [ipxe/ipxe#1235] [General]
We have one project with onboard Intel 82599 NIC chip can't boot from the iPXE chainloading environment, but it can boot from the PXE environment, we can't figure out what is the reason, please help to provide the suggestions, thanks.
![图片](https://github.com/ipxe/ipxe/assets/171772999/584f533d-4d7f-4d87-8906-82b22f7e663d)

**Comment by NiKiZe** [comment]
Have you followed the link in the message, and checked the suggestions there?

You are using legacy undionly, I assume this is `undionly.kpxe` ?
You also seem to be using an old version.

After checking the linked page, and if that doesn't help you resolve the problem, try again, and after `dhcp` run `ifstat` again and check TXE and RXE values.

**Reply by JetYu1234** [reply]
Thanks for your comments, NiKiZe. Yes, our server is using legacy mode, but we download the ipxe package from https://github.com/ipxe/ipxe, and follow the steps to make then generated the undionly.kpxe, after that the server show the boot result as picture.

**Comment by goodboy12ym** [comment]
Hi, We have used the latest ipxe source code and build the image, when we are using undionly.kpxe image, we found the server still failing in DHCP discovery,  we can see `Link:up, TX:4 TXE:1 RX:0 RXE:0`  in ifstat.

But when we tried a older one, ipxe.pxe. the ifstat become `Link:up, TX:4 TXE:1 RX:27 RXE:16` and the `dhcp` command succeeded.

Do u know whats the reason for why the undionly.kpxe is not working but ipxe.pxe can work? Currently we are only seen this type NIC is failing.

**Reply by NiKiZe** [reply]
undionly.kpxe uses the undi stack from your NIC ROM/BIOS.
ipxe.x however have several native drivers, and no undi driver.

The `ifstat` output does show which driver is in use for the NIC.
If undionly.kpxe doesn't work, it often indicates that the ROM/BIOS has a buggy UNDI implementation.

For more help, post the full output of ifstat, also include version information of iPXE for the different boots.

**Reply by goodboy12ym** [reply]
Hi NiKiZe, here is the output we have for ifstat when I boot with **undionly.kpxe** on one server:
<img width="608" alt="Screenshot 2024-08-08 at 5 36 47 PM" src="https://github.com/user-attachments/assets/6a51a069-cb9f-46eb-a2da-70bb6db87aa6">

With **ipxe.pxe** on same host, its following:
<img width="551" alt="Screenshot 2024-08-08 at 4 31 06 PM" src="https://github.com/user-attachments/assets/33c01344-420f-4453-aaf5-ffc6745b2a02">

Both versions are 1.21.1+.
Please let me know if u need any other information.

**Reply by NiKiZe** [reply]
Yes, something with the undistack is broken, try to upgrade your BIOS or the ROM on NIC.

The version number has (g.....) This is the commit that was used which is an important part to know what actually you are running.
